Generate PDF report for cypress + environment management improvements (#1079)
opendevstack/ods-quickstarters: e2e-cypress/files/README.md

## Local development

Run `npm start` to develop the e2e tests. The tests will automatically rebuild and run, if you change any of the source files. Ideally the test will run against a local instance of the front end, e.g. `http://localhost:4200` for an Angular app. This destination is configurable in the `cypress.config.ts` file.

## Reports
From [Merging reports across spec files](https://docs.cypress.io/guides/tooling/reporters#Merging-reports-across-spec-files): each spec file is processed completely separately during each cypress run execution. Thus each spec run overwrites the previous report file. To preserve unique reports for each spec file, use the `[hash]` in the `mochaFile` filename.

In order to generate one xml report per test type (installation, integration and acceptance) we use the junit-report-merger tool. See also the `junit-...-report` tasks in `package.json`.

## E2e test user authentication

> With Cypress 12 version is now available `cy.origin()` that allows you to handle redirections. This functionality eases the login handling.

Reviewer: ```suggestion
Starting version 12 of Cypress `cy.origin()` allows you to handle redirections. This functionality eases the login handling.
```

See `./support/login-functions.ts` for a generic login example.

> In order to load your testing user's credentials for this login with SSO, you need to create a secret in OpenShift with the label `credential.sync.jenkins.openshift.io=true` and import them as environment variables using the withCredentials block to keep them secure. Find an example on how to do this in the `Jenkinsfile`.

Reviewer: ```suggestion
In order to load your testing user's credentials for this login with SSO, you need to create a secret in OpenShift with the label `credential.sync.jenkins.openshift.io=true` and import them as environment variables using the `withCredentials` block to keep them secure. Find an example on how to do this in the `Jenkinsfile`.
```
